pkg/replicate/replicator.go

```diff
@@ -243,3 +235,32 @@ func RunReplicate(
 
 	return nil
 }
+
+func newMetaFetcher(
+	logger log.Logger,
+	fromBkt objstore.InstrumentedBucket,
+	reg prometheus.Registerer,
+	minTime,
+	maxTime thanosmodel.TimeOrDurationValue,
+	concurrency int,
+	ignoreMarkedForDeletion bool,
+) (*thanosblock.MetaFetcher, error) {
+	filters := []thanosblock.MetadataFilter{
+		thanosblock.NewTimePartitionMetaFilter(minTime, maxTime),
+	}
+	if ignoreMarkedForDeletion {
+		filters = append(filters, thanosblock.NewIgnoreDeletionMarkFilter(logger, fromBkt, 0, concurrency))
+	}
+	fetcher, err := thanosblock.NewMetaFetcher(
```

Comment:
We can just return `thanosblock.NewMetaFetcher(...)`

Reply:
Right, the error was previously wrapped which was the reason it was not returned right a way, but I eventually wrapped it where the func is called and forget about this. Thanks